readme: clarify host lifecycle behaviour (#3482)
bottlerocket-os/bottlerocket: README.md

* They run in a separate instance of containerd than the one used for orchestrated containers like Kubernetes pods.
* They're not updated automatically. You need to update the `source` and commit those changes.
* If you set `superpowered` to true, they'll essentially have root access to the host.

> * If the container exits (whether successfully or not), it will always restart after 45 seconds

Reviewer: Thanks, this is good clarification. A minor suggestion on the wording, but looks good!

```suggestion
* If the container exits for any reason, Bottlerocket will attempt to restart it after 45 seconds.
```
